source/FreeRTOS_IP_Utils.c

```diff
@@ -291,8 +289,12 @@ NetworkBufferDescriptor_t * pxUDPPayloadBuffer_to_NetworkBuffer( const void * pv
 BaseType_t xIsCallingFromIPTask( void )
 {
     BaseType_t xReturn;
+    TaskHandle_t xCurrentHandle, xCurrentIPTaskHandle;
+
+    xCurrentHandle = xTaskGetCurrentTaskHandle();
+    xCurrentIPTaskHandle = FreeRTOS_GetIPTaskHandle();
 
-    if( xTaskGetCurrentTaskHandle() == FreeRTOS_GetIPTaskHandle() )
+    if( xCurrentHandle == xCurrentIPTaskHandle )
```

Comment:
Is this to fix a particular MISRA violation?

Reply:
Yes, MISRA check does not like the expression inside evaluation.  It might have unexpected result if the execution order might change the result.

Reply:
Can you clarify the Rule #?

Reply:
I'm guessing that xTaskGetCurrentTaskHandle performs multiple volatile access operations which MISRA considers a potentially persistent side-effect. Seems very unlikely to be a problem though.